build_config/lib/src/parse.dart

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:yaml/yaml.dart';
+
+import 'build_config.dart';
+
+/// Supported values for the `platforms` attribute.
+const _allPlatforms = const [
+  _vmPlatform,
+  _webPlatform,
+  _flutterPlatform,
+];
+const _vmPlatform = 'vm';
+const _webPlatform = 'web';
+const _flutterPlatform = 'flutter';
+
+const _targetOptions = const [
+  _builders,
+  _default,
+  _dependencies,
+  _excludeSources,
+  _generateFor,
+  _platforms,
+  _sources,
+];
+const _builders = 'builders';
+const _default = 'default';
+const _dependencies = 'dependencies';
+const _excludeSources = 'exclude_sources';
+const _generateFor = 'generate_for';
+const _platforms = 'platforms';
+const _sources = 'sources';
+
+const _builderOptions = const [
+  _builderFactories,
+  _import,
+  _buildExtensions,
+  _target,
+  _autoApply,
+  _requiredInputs,
+  _isOptional,
+  _buildTo,
+];
+const _builderFactories = 'builder_factories';
+const _import = 'import';
+const _buildExtensions = 'build_extensions';
+const _target = 'target';
+const _autoApply = 'auto_apply';
+const _requiredInputs = 'required_inputs';
+const _isOptional = 'is_optional';
+const _buildTo = 'build_to';
+
+BuildConfig parse(
```

Comment:
nit: this is a bit generic for a top level method... `parseFromYaml` or `parseConfig`?